### PR TITLE
Quiz answers [medium-sized-pull-request-im-sorry]

### PR DESCRIPTION
**Pull request by deadlybutter**
### What does this PR do?
This PR implements all of the logic for the quiz functionality, _except_ for the conversion step & template variables for rendering the different quiz outcomes. That'll be the next PR, and I left in some placeholder that kinda outlines the direction that will go in.

The redux store in this PR is setup to keep track of what points to award users for each question in each quiz. If people answer all of the questions, it lets them see the response. 

It got kinda large because there is a ton of boiler plate to setup, sorry!

### CODE REVIEW

File: resources/assets/actions/index.js [hunk 1]
```diff
@@ -75,6 +75,13 @@ export const OPEN_MODAL = 'OPEN_MODAL';
 export const CLOSE_MODAL = 'CLOSE_MODAL';
 export * from './modal';
 
+export const QUIZ_INIT = 'QUIZ_INIT';
+export const PICK_QUIZ_ANSWER = 'PICK_QUIZ_ANSWER';
+export const VIEW_QUIZ_RESULT = 'VIEW_QUIZ_RESULT';
+export const COMPARE_QUIZ_ANSWER = 'COMPARE_QUIZ_ANSWER';
+export const QUIZ_ERROR = 'QUIZ_ERROR';
```
Comment (deadlybutter) [inline resources/assets/actions/index.js]: Should we track this? Couldn't hurt I guess

File: resources/assets/components/Quiz/Quiz.js [hunk 2]
```diff
@@ -1,20 +1,34 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import QuizInitializr from './QuizInitializr';
 import Markdown from '../Markdown';
 import Question from './Question';
 
-const Quiz = ({ fields }) => (
+const Quiz = ({ id, fields, data, viewQuizResult, pickQuizAnswer, quizInit }) => (
   <div className="quiz">
+    <QuizInitializr quizId={id} quizInit={quizInit} />
```
Comment (deadlybutter) [inline resources/assets/components/Quiz/Quiz.js]: Hm, I probably don't need this in retrospect, we have default values...

Comment (deadlybutter) [inline resources/assets/components/Quiz/Quiz.js]: Oh, wait, yeah I remember I do need this. When the page reloads after conversion we'll want this to fire and make sure the previous answers are injected into the state.

File: resources/assets/actions/quiz.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import {
+  PICK_QUIZ_ANSWER,
+  COMPARE_QUIZ_ANSWER,
+  VIEW_QUIZ_RESULT,
+  QUIZ_INIT,
+  QUIZ_ERROR,
+} from '../actions';
+
+export function quizInit(quizId) {
+  return { type: QUIZ_INIT, quizId };
```
Comment (DFurnes) [inline resources/assets/actions/quiz.js]: What's this used for? I'm a little wary of having many `_INIT` style events.

File: resources/assets/components/Quiz/Quiz.js (hunk 2 above)
Comment (DFurnes) [inline resources/assets/components/Quiz/Quiz.js]: Hmm, I'm not a huge fan of this pattern. I think if we need to do stuff like this, let's make a wrapping component that handles lifecycle events and then renders into a functional component.

Also `QuizInitializr` → `QuizInitializer`! ✏️

File: resources/assets/components/Quiz/Quiz.js [hunk 4]
```diff
@@ -1,20 +1,34 @@
...
     <h1 className="quiz__title">{fields.title}</h1>
-    <Markdown>{fields.introduction || ''}</Markdown>
-    {(fields.questions || []).map(question => (
-      <Question key={question.id} {...question} />
+    {data.shouldSeeResult ? null : (
+      <Markdown>{fields.introduction || ''}</Markdown>
+    )}
+    {data.shouldSeeResult ? null : (fields.questions || []).map(question => (
```
Comment (DFurnes) [inline resources/assets/components/Quiz/Quiz.js]: For `(fields.questions || [])` and company, can we use default prop values?

File: resources/assets/components/Quiz/Quiz.js (hunk 2 above)
Comment (deadlybutter) [inline resources/assets/components/Quiz/Quiz.js]: Yeah I think that makes sense 👍

File: resources/assets/components/Quiz/Quiz.js (hunk 4 above)
Comment (deadlybutter) [inline resources/assets/components/Quiz/Quiz.js]: Yeah!

**Comment by deadlybutter** [comment]
@DFurnes Hows that?

**Pull request merged**